gtsam/nonlinear/nonlinear.i

```diff
@@ -522,6 +522,14 @@ virtual class DoglegParams : gtsam::NonlinearOptimizerParams {
   void setVerbosityDL(string verbosityDL) const;
 };
 
+#include <gtsam/nonlinear/GncParams.h>
+template<BaseOptimizerParameters>
```

Comment:
Nitpick: Use a smaller uppercase name like `PARAMS` and provide either a default via the template list `PARAMS = {SomeValue}` or use the typedef to declare on (I prefer the former).

Without a template list of typedef, the python module import should error out.

Reply:
You're right, i checked and it compiled but import didn't work.

I updated the templates, let me know if this is what you had in mind @varunagrawal. Thanks.

Reply:
weird, even with the updated templates we still get
```bash
======================================================================
ERROR: test_NonlinearOptimizer (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_NonlinearOptimizer
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/home/runner/work/gtsam/gtsam/python/gtsam/tests/test_NonlinearOptimizer.py", line 18, in <module>
    from gtsam import (DoglegOptimizer, DoglegParams,
ImportError: cannot import name 'GncParams'
```

Reply:
I'm guessing this is because we were missing GTSAM_EXPORT calls, and because the class wasn't marked as virtual in the .i files. Do we have those steps documented somewhere in `gtwrap`, along with why they are required? I'm a bit confused about that.

Reply:
Thanks @varunagrawal, looks like all the tests pass now.